### PR TITLE
[musiclibrary] Made top albums sortable by play count

### PR DESCRIPTION
**Pull request by razzeee**
- Added "play count" sorting for "top 100 albums"
- Made sure the album song sorting gets saved if you choose the "play count" sorting there

Fixes http://trac.kodi.tv/ticket/16207

### CODE REVIEW

**Comment by Razzeee** [comment]
Has the same problem that is discussed in https://github.com/xbmc/xbmc/pull/7841

**Comment by Paxxi** [comment]
just a random comment. Keep up the work @Razzeee even if we're being tough on you sometimes. Musiclibrary needs a lot of love and don't hesitate to yell for assistance :)

**Comment by Razzeee** [comment]
@Paxxi thank you. You need to be tough otherwise nothing will improve. 

And it's hard to reach out most of the times

**Comment by Razzeee** [comment]
Rebased, just for the fun of it

**Comment by hudokkow** [comment]
jenkins build this please

**Comment by Razzeee** [comment]
Can somebody review this? 
@mkortstiege @Montellese

**Comment by mkortstiege** [comment]
I like trains. Apart from that i looks good to me. @MartijnKaijser, @zag2me what's your opinion on this one?

**Comment by MartijnKaijser** [comment]
makes sense to have it as option. i like planes

**Comment by zag2me** [comment]
For my limited understanding of the code, it looks like this is a new sort option for albums that has been missing for a long time! So + 1 for me. I like beer ;)

File: xbmc/music/GUIViewStateMusic.cpp [hunk 1]
```diff
@@ -280,6 +282,7 @@ CGUIViewStateMusicDatabase::CGUIViewStateMusicDatabase(const CFileItemList& item
       AddSortMethod(SortByRating, 563, LABEL_MASKS("%T - %A", "%R"));  // Title - Artist, Rating
       AddSortMethod(SortByYear, 562, LABEL_MASKS("%T - %A", "%Y")); // Title, Artist, Year
       AddSortMethod(SortByDateAdded, 570, LABEL_MASKS("%A - %T", "%a"));  // Title - Artist, DateAdded | empty, empty
```
Comment (Montellese) [inline xbmc/music/GUIViewStateMusic.cpp]: Just noticed in this diff that the `SortByYear` has the label mask in the wrong order.

File: xbmc/music/GUIViewStateMusic.cpp [hunk 2]
```diff
@@ -193,6 +193,8 @@ CGUIViewStateMusicDatabase::CGUIViewStateMusicDatabase(const CFileItemList& item
       AddSortMethod(SortByYear, 562, LABEL_MASKS("%F", "", strAlbumLeft, strAlbumRight));
       // album date added
       AddSortMethod(SortByDateAdded, sortAttribute, 570, LABEL_MASKS("%F", "", strAlbumLeft, strAlbumRight));  // Filename, empty | Userdefined, Userdefined
+      // play count
+      AddSortMethod(SortByPlaycount, 567, LABEL_MASKS("%F", "", strAlbumLeft + " - " + strAlbumRight, "%V"));  // Filename, empty | Userdefined - Userdefined, Play count
```
Comment (Montellese) [inline xbmc/music/GUIViewStateMusic.cpp]: Once again I'd like to point out that we should stick to the old way of putting together the label mask until it has been decided how to handle this for all affected sort methods.

**Comment by Montellese** [comment]
The added feature is fine IMO. I left one comment that is not directly related to this PR and one that is.

File: xbmc/music/GUIViewStateMusic.cpp (hunk 1 above)
Comment (Razzeee) [inline xbmc/music/GUIViewStateMusic.cpp]: Not sure what you mean, I think we have different plans/understanding for this. Can you elaborate?

Comment (Montellese) [inline xbmc/music/GUIViewStateMusic.cpp]: All `SortByFoo` methods here use `%T - %A` as the label format but `SortByDataAdded` uses `%A - %T`. It's not part of this PR but I just noticed that it's inconsistent.

Comment (Razzeee) [inline xbmc/music/GUIViewStateMusic.cpp]: Ah, didn't get that due to you saying ```SortByYear```

File: xbmc/music/GUIViewStateMusic.cpp (hunk 2 above)
Comment (Razzeee) [inline xbmc/music/GUIViewStateMusic.cpp]: Changed it back.

**Comment by Montellese** [comment]
Fine with me.

**Comment by Razzeee** [comment]
Up to speed again.

**Comment by Razzeee** [comment]
@Montellese 
your button
